### PR TITLE
Rework CMake to call waf directly

### PR DESCRIPTION
**Pull request by OznOg**
### What does this PR do?
Rework cmake build of libaubio so that it may work everywhere (hopefully)

### Closes Issue(s)

nop

### Motivation
I've been having problems to compile master

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -91,20 +92,23 @@ if(NOT FFTW3_FOUND OR NOT BLAS_FOUND)
...
-    UPDATE_COMMAND ""
-	TMP_DIR tmp
-	CONFIGURE_COMMAND "${CMAKE_SOURCE_DIR}/3rdparty/build_aubio.sh" ${ACCELERATE_COMMAND} <INSTALL_DIR> ${AUBIO_CFLAGS}
-	BUILD_COMMAND ""
+        BINARY_DIR ${CMAKE_BINARY_DIR}/3rdparty/aubio-build/
+        INSTALL_DIR ${CMAKE_BINARY_DIR}/3rdparty/aubio-install/
+        UPDATE_COMMAND ""
+	TMP_DIR ${CMAKE_BINARY_DIR}/3rdparty/aubio-build/tmp/
+	CONFIGURE_COMMAND ""
+	BUILD_COMMAND cmake -E env CFLAGS=${AUBIO_CFLAGS} ${CMAKE_SOURCE_DIR}/3rdparty/aubio/waf
```
Comment (Baklap4) [inline CMakeLists.txt]: I'd change `cmake` to `${CMAKE_COMMAND}`
As per cmake docs;
> This is the full path to the CMake executable cmake(1) which is useful from custom commands that want to use the cmake -E option for portable system commands. (e.g. /usr/local/bin/cmake)

Comment (OznOg) [reply CMakeLists.txt]: yes you're right

File: CMakeLists.txt [hunk 2]
```diff
@@ -91,26 +92,29 @@ if(NOT FFTW3_FOUND OR NOT BLAS_FOUND)
...
+        BINARY_DIR ${CMAKE_BINARY_DIR}/3rdparty/aubio-build/
+        INSTALL_DIR ${CMAKE_BINARY_DIR}/3rdparty/aubio-install/
+        UPDATE_COMMAND ""
+	TMP_DIR ${CMAKE_BINARY_DIR}/3rdparty/aubio-build/tmp/
+	CONFIGURE_COMMAND ""
+	BUILD_COMMAND ${CMAKE_COMMAND} -E env CFLAGS=${AUBIO_CFLAGS} ${CMAKE_SOURCE_DIR}/3rdparty/aubio/waf
+                  ${ACCELERATE_COMMAND} --disable-tests --notests --verbose
+                  --enable-fftw3f --disable-sndfile --disable-avcodec --disable-double
+                  --disable-samplerate --disable-docs --disable-wavread --disable-wavwrite
+                  --disable-tests --notests --disable-examples --disable-apple-audio
```
Comment (Lord-Kamina) [inline CMakeLists.txt]: You shouldn't hard disable audio engines here.

File: CMakeLists.txt [hunk 3]
```diff
@@ -91,26 +92,29 @@ if(NOT FFTW3_FOUND OR NOT BLAS_FOUND)
...
+                  --disable-fat --disable-jack --libdir=lib
```
Comment (Lord-Kamina) [inline CMakeLists.txt]: Same as above

File: CMakeLists.txt (hunk 2 above)
Comment (OznOg) [reply CMakeLists.txt]: those are options from the original (upstream) shell script; can you be more specific on what are the flags you want to see here?

File: game/CMakeLists.txt [hunk 4]
```diff
@@ -170,8 +170,9 @@ if(WIN32)
   target_link_libraries(performous wsock32 ws2_32)
 endif()
 
-target_link_libraries(performous ced)
-list(APPEND LIBS ${AUBIO_INSTALL_DIR}/lib/libaubio.a)
+target_link_libraries(performous ced aubio)
+target_link_directories(performous BEFORE PRIVATE ${AUBIO_INSTALL_DIR}/lib/)
```
Comment (Baklap4) [inline game/CMakeLists.txt]: target_link_directories is only available only available from cmake 3.13 or higher.

Comment (OznOg) [reply game/CMakeLists.txt]: that is bad news, we'll have to do it in a ugly way then

Comment (OznOg) [reply game/CMakeLists.txt]: So, do we eventually need aubio to be built on windows?

Comment (Baklap4) [reply game/CMakeLists.txt]: Eventually yeah, however we're not using the pre-build crosscompiled libraries

**Comment by OznOg** [comment]
so where do we stand with this? Is it worth keeping it?

**Comment by Baklap4** [comment]
Thanks for the PR @OznOg !

**Pull request merged**